lib/uktt/commodity.rb

```diff
@@ -3,43 +3,20 @@ module Uktt
   class Commodity < Base
     RESOURCE_PATH = 'commodities'.freeze
 
-    attr_accessor :commodity_id, :response
-
-    def initialize(opts = {})
-      @commodity_id = opts[:commodity_id] || nil
-      Uktt.config.merge(opts)
-      @config = Uktt.config
-      @response = nil
+    def retrieve(commodity_id)
+      fetch "#{RESOURCE_PATH}/#{commodity_id}.json"
     end
 
-    def retrieve
-      return '@commodity_id cannot be nil' if @commodity_id.nil?
-
-      fetch "#{RESOURCE_PATH}/#{@commodity_id}.json"
-    end
-
-    def changes
-      return '@commodity_id cannot be nil' if @commodity_id.nil?
-
-      fetch "#{RESOURCE_PATH}/#{@commodity_id}/changes.json"
+    def changes(commodity_id)
+      fetch "#{RESOURCE_PATH}/#{commodity_id}/changes.json"
     end
 
     def find(id)
-      return '@response is nil, run #retrieve first' unless @response
-
-      response = @response.included.select do |obj|
-        obj.id === id || obj.type === id
-      end
-      response.length == 1 ? response.first : response
+      @response.included.find { |obj| obj.id == id }
     end
 
     def find_in(arr)
-      return '@response is nil, run #retrieve first' unless @response
-
-      response = @response.included.select do |obj|
-        arr.include?(obj.id)
-      end
-      response.length == 1 ? response.first : response
+      @response.included.find { |obj| arr.include?(obj.id) }
```

Comment:
Same question as: https://github.com/trade-tariff/uktt/pull/6/files#r594173830

Reply:
Removed as not used